File: src/main/java/me/sangbo/work/Rev2.java
package me.sangbo.work;

import com.rabbitmq.client.*;
import me.sangbo.ConnectionUtils;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

/**
 * Created by 桑博 on 2018/6/13.
 */
public class Rev2 {


  public static final String QUEUE_NAME= "queue_work_sangbo";
  public static void main(String[] args) throws IOException, TimeoutException {

    Connection connection = ConnectionUtils.getConnection();
    Channel channel = connection.createChannel();

    channel.queueDeclare(QUEUE_NAME,false,false,false,null);

    DefaultConsumer defaultConsumer = new DefaultConsumer(channel){
      @Override
      public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) throws IOException {

        System.out.println("rev2收到的消息："+new String(body,"utf-8"));
        try {
          Thread.sleep(2000);
        } catch (InterruptedException e) {
          e.printStackTrace();
        }
      }
    };

    channel.basicConsume(QUEUE_NAME,true,defaultConsumer);



  }

}

File: src/main/java/me/sangbo/roulting/Rev2.kt
package me.sangbo.roulting

import com.rabbitmq.client.AMQP
import com.rabbitmq.client.Channel
import com.rabbitmq.client.DefaultConsumer
import com.rabbitmq.client.Envelope
import me.sangbo.ConnectionUtils

/**
 * Created by 桑博 on 2018/6/14.
 */
private const val QUEUE_NAME  = "queue_roulting_2_sangbo"
private const val EXCHANGE_NAME = "exchange_sangbo_direct"
private const val ROULTING_KEY = "info";
fun main(args: Array<String>) {

    val connection = ConnectionUtils.getConnection()
    val channel = connection.createChannel()
    channel.queueDeclare(QUEUE_NAME,false,false,false,null);
    channel.exchangeDeclare(EXCHANGE_NAME,"direct")
    channel.queueBind(QUEUE_NAME, EXCHANGE_NAME, ROULTING_KEY);
    channel.basicQos(1)

    val consumer = object : DefaultConsumer(channel){
        override fun handleDelivery(consumerTag: String?, envelope: Envelope?, properties: AMQP.BasicProperties?, body: ByteArray?) {

            println("rev2："+ String(body!!));
            Thread.sleep(1000);
            channel.basicAck(envelope!!.deliveryTag,false)
        }
    }

    val autoAck = false;
    channel.basicConsume(QUEUE_NAME,autoAck,consumer);



}
